rust/lance-encoding/src/buffer.rs

```diff
@@ -114,6 +173,110 @@ impl LanceBuffer {
             }),
         }
     }
+
+    /// Make an owned copy of the buffer (always does a copy of the data)
+    pub fn deep_copy(&self) -> Self {
+        match self {
+            Self::Borrowed(buffer) => Self::Owned(buffer.to_vec()),
+            Self::Owned(buffer) => Self::Owned(buffer.clone()),
```

Comment:
if the `buffer` is already an owned `Vec<u8>`, I don't see the necessity of `clone()` here, what am I missing here? maybe better to return a `Vec<u8>`?

Reply:
oh, I see it now, you're returning a newly constructed `LanceBuffer`

Reply:
Yep.  I tried to make it so that a deep copy is data is not easy to do automatically.  E.g. it does not implement `Clone` (but we have `try_clone` and `borrow_and_clone`)

So this is the "ok, I really do want to do a full copy of the data".